course5/homework03/homework03.cc

```diff
@@ -1,18 +1,68 @@
 #include <cstdio>
+#include <string>
 
-// TODO your code goes here
+/***
+Imagine that you are a young C++ developer, and your manager, for some reason, forbids std::cout. The only I/O function you are allowed to use to produce the output, is std::printf from <cstdio>
+
+You are not happy about it; you are using C++ after all, and you want to take advantage of the type-safety and convenience it offers.
+
+1. Write a library that provides you with “print”, a function you can use on a selection of types: int, bool, string, float, double etc. The function would use std::printf under the covers to print the value.
+2. For floating-point values, the function could take a second, optional parameter, integer, that specifies the precision with which the number is printed.
+***/
+namespace printer
+{
+	/***
+	provides overloaded function to print various types to stdout
+	***/
+	void print(int input)
+	{
+		std::printf("%i", input);
+	}
+
+	void print(double input, int precision = 0)
+	{
+		std::string format_str = "%." + std::to_string(precision) + "f";
+		const char * format_ptr = format_str.c_str();
+		std::printf(format_ptr, input);
+	}
+
+	void print(bool input)
+	{
+		if (input == 1)
```

Comment:
Comparing boolean to integer is ugly.
"if" expects a boolean expression, so you can just say: ```if(input)```
If you need explicit boolean literals, use ```true``` and ```false```.